addons/web/static/src/views/pivot/pivot_renderer.js

```diff
@@ -84,13 +84,24 @@ export class PivotRenderer extends Component {
      */
     getFormattedValue(cell) {
         const field = this.model.metaData.measures[cell.measure];
+        const fieldInfo = {};
+        const fieldAttrs = this.model.metaData.fieldAttrs[cell.measure];
+        if (fieldAttrs && fieldAttrs.options && Object.keys(fieldAttrs.options).length) {
+            fieldInfo.options = fieldAttrs.options;
+        }
         let formatType = this.model.metaData.widgets[cell.measure];
         if (!formatType) {
             const fieldType = field.type;
             formatType = ["many2one", "reference"].includes(fieldType) ? "integer" : fieldType;
         }
         const formatter = formatters.get(formatType);
-        return formatter(cell.value, field);
+        const formatOptions = {};
+        if (fieldInfo.options && formatter.extractOptions) {
+            fieldInfo.attr = {};
+            Object.assign(formatOptions, formatter.extractOptions(fieldInfo));
+        }
+        Object.assign(formatOptions, field);
```

Comment:
formatOptions.field = field;
--> tests should be added to cover problematic cases